spring-integration-core/src/main/java/org/springframework/integration/dsl/IntegrationFlowDefinition.java

```diff
@@ -213,6 +213,18 @@ public B channel(MessageChannel messageChannel) {
 	}
 
 	/**
+	 * Populate a {@link MessageChannel} instance
+	 * at the current {@link IntegrationFlow} chain position using the {@link Channels}
+	 * factory fluent API.
+	 * @param channels the {@link Function} to use.
+	 * @return the current {@link IntegrationFlowDefinition}.
+	 */
+	public B channel(Function<Channels, MessageChannelSpec<?, ?>> channels) {
```

Comment:
No, we can't use this. Because there is `.channel(MessageChannel)` variant and the last one is `@FunctionalInterface` as well. Therefore ambiguity. That's why I deleted this method and therefore `Channels` factory.

I won't mind if I am missing something...